[All]: Remove support for Legacy (25.6) [STUD-74234]

viogroza opened the pull request:
Remove legacy support for Python
Remove legacy support for Java
Remove legacy support for Database
Remove legacy support for Credentials
Remove legacy support for Cryptography
Remove legacy support for FTP
Switch from NUnit to xUnit for all 

Remove support for Python 2.7, 3.4 and 3.5
Clean net5.0 and netcoreapp from directory.build.targets

sorinconstantin commented on Activities/Python/UiPath.Python.Packaging/UiPath.Python.Packaging.csproj:
```diff
@@ -165,11 +149,11 @@
 	</ItemGroup>
 
-  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net461' AND $(ConfigurationName) == Debug ">
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net6.0' AND $(ConfigurationName) == Debug ">
```
the condition suggested the command should be executed in case TargetFramework was net461.
so, why do we execute it now for net6.0?

viogroza replied:
this condition removes the package before the generating the package..
from what I saw.. it functions as below:
1. build for all platforms (net461, net6.0, net6.0-windows .. which of them are present)
2. post build for net461 delete the old nuget package (notice that only for debug the script is ran)
3. generate nuget package

viogroza committed:
Python: Remove support for Legacy (25.6) [STUD-74234]

Remove legacy support for Python
Remove legacy support for Java
Remove legacy support for Database
Remove legacy support for Credentials
Remove legacy support for Cryptography
Remove legacy support for FTP

viogroza committed:
Python: Remove support for Legacy (25.6) [STUD-74234]

Remove legacy support for Python
Remove legacy support for Java
Remove legacy support for Database
Remove legacy support for Credentials
Remove legacy support for Cryptography
Remove legacy support for FTP
Update a few dependencies

viogroza committed:
Python: Remove support for Legacy (25.6) [STUD-74234]

Update more dependencies
Centralize dependencies for testprojects

sorinconstantin commented on Activities/Database/ConnectionDialog/UiPath.Data.ConnectionUI.Dialog/UiPath.Data.ConnectionUI.Dialog.csproj:
```diff
@@ -41,15 +41,9 @@
   <ItemGroup Condition="'$(TargetFramework)' == 'net6.0-windows'">
     <PackageReference Include="Oracle.ManagedDataAccess.Core">
-      <Version>3.21.90</Version>
+      <Version>23.8.0</Version>
```
update to latest

viogroza replied:
will do

sorinconstantin commented on Activities/Database/UiPath.Database.Activities.Design/UiPath.Database.Activities.Design.csproj:
```diff
@@ -11,23 +11,12 @@
     <PackageReference Include="UiPath.Workflow" />
     <PackageReference Include="System.Activities.Core.Presentation" />
-    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="23.5.0" PrivateAssets="All" />
+    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="23.8.0" PrivateAssets="All" />
```
here as well?

viogroza replied:
here is latest

viogroza committed:
All: Remove support for Legacy (25.6) [STUD-74234]

Update more dependencies
Centralize dependencies for testprojects

alexandru-petre commented on Activities/Credentials/UiPath.Credentials.Packages/UiPath.Credentials.Packages.csproj:
```diff
@@ -62,13 +58,13 @@
 	</ItemGroup>
 
-	<Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net461' AND $(ConfigurationName) == Debug ">
+	<Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net6.0-windows' AND $(ConfigurationName) == Debug ">
```
Based on the old condition, this was executed only on .net461, so it should rather be deleted.

viogroza replied:
it should not be deleted
it deletes the old package
the idea was to be executed only once (to delete the package)
if you check you will notice that it was executed only on debug

alexandru-petre commented on Activities/Database/ConnectionDialog/UiPath.Data.ConnectionUI.Dialog/UiPath.Data.ConnectionUI.Dialog.csproj:
```diff
@@ -41,15 +41,9 @@
+      <Version>23.8.0</Version>
     </PackageReference>
     <PackageReference Include="System.Activities">
       <Version>1.0.0-20201117-04</Version>
```
This version (1.0.0-20201117-04) seems kind of old too

viogroza replied:
I will remove the version; this will use the version from directory.build.targets

viogroza replied:
I removed all this block since the package is referenced before

alexandru-petre commented on Activities/Database/UiPath.Database.Activities.Packaging/UiPath.Database.Activities.Packaging.csproj:
```diff
@@ -129,7 +123,7 @@
   </ItemGroup>
 
-  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net461' AND $(ConfigurationName) == Debug ">
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net6.0' AND $(ConfigurationName) == Debug ">
```
Seems to be a block that needs to be deleted.

viogroza replied:
it's needed (I mentioned the reason in a previous comment)

alexandru-petre commented on Activities/Java/UiPath.Java.Activities.Packaging/UiPath.Java.Activities.Packaging.csproj:
```diff
@@ -82,13 +82,13 @@
 	</ItemGroup>
 
-	<Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net461' AND $(ConfigurationName) == Debug ">
+	<Target Name="PostBuild" AfterTargets="PostBuildEvent" Condition=" '$(TargetFramework)' == 'net6.0' AND $(ConfigurationName) == Debug ">
```
Same

viogroza replied:
same

alexandru-petre commented on Activities/Shared/Test.build.targets:
```diff
@@ -1,15 +1,12 @@
-    <PackageReference Include="Shouldly" Version="3.0.2" />
-    <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+    <PackageReference Include="Moq" Version="4.20.72" />
```
Moq 4.20 is the version where the privacy shitstorm started, we should stay below it if we can.

viogroza replied:
looking into it

viogroza replied:
I removed this file since the inclusion of these settings is done now in Directory.build.targets

alexandru-petre commented on Activities/Directory.build.targets:
```diff
@@ -67,18 +67,26 @@
-		<PackageReference Include="xunit" Version="2.4.1" />
-		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
-		<PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+		<PackageReference Include="Moq" Version="4.20.72" />
```
Same

viogroza replied:
use 4.18.4

sorinconstantin commented on Activities/Database/UiPath.Database.Tests/DatabaseConnectionTests.cs:
```diff
@@ -25,9 +24,11 @@ public void TestConnect()
+
+            WorkflowInvoker.Invoke(connectActivity, TimeSpan.FromSeconds(30));
+            //var host = new WorkflowInvokerTest(connectActivity);
+            //var ex = Record.Exception(() => host.TestActivity());
```
these tests assert nothing. Do we keep them?

viogroza replied:
yes, we want to keep them
even before the change, they asserted that no exception was thrown
now, just run the workflow and assume that it not throws (before the condition was that there is no exception Assert.Null(ex);

viogroza committed:
All: Remove support for Legacy (25.6) [STUD-74234]

Address review findings
Remove unused file
Switch from NUnit to xUnit
Fix display name for runtime tests

sorinconstantin commented on Activities/Database/ConnectionDialog/UiPath.Data.ConnectionUI.Dialog/UiPath.Data.ConnectionUI.Dialog.csproj:
```diff
@@ -16,19 +16,22 @@
-    <Reference Include="System.Activities.Presentation" />
 
+  <ItemGroup>
+    <PackageReference Include="Oracle.ManagedDataAccess.Core" Version="23.8.0" PrivateAssets="All" />
```
is 23.8.0 the right version

sorinconstantin commented on Activities/Shared/Test.build.targets:
```diff
@@ -1,15 +0,0 @@
-
-    <PackageReference Update="Microsoft.Activities.UnitTesting" Version="2.0.6.9" />
-  </ItemGroup>
-</Project>
```
is this not needed anymore?

viogroza replied:
I removed the file altogether

viogroza committed:
All: Remove support for Legacy (25.6) [STUD-74234]

Clean net5.0 and netcoreapp from directory.build.targets
Add System.Activities.ViewModels dependency for FTP

viogroza commented:
I will update runtime cryptography tests in another PR (it fails on develop even before these changes)
